bip-0352.mediawiki

```diff
@@ -0,0 +1,471 @@
+
+==== Scanning ====
+
+If each of the checks in ''[[#scanning-transactions|Scanning transactions]]'' passes, the receiving wallet must:
+
+* Generate the ''outpoints_hash'', using the method described above
+* Let ''A = A<sub>0</sub> + A<sub>1</sub> + … A<sub>n</sub>'', where each ''A<sub>i</sub>'' is the public key of an input from the ''[[#inputs-for-shared-secret-derivation|Inputs For Shared Secret Derivation]]'' list
+* Let ''ecdh_shared_secret = outpoints_hash·b<sub>scan</sub>·A''
+* Check for outputs:
+** Let ''outputs_to_check = the taproot output key from each unspent taproot output in the transaction''
+** Starting with ''n = 0'':
+*** Let ''t<sub>n</sub> = sha256(ser<sub>P</sub>(ecdh_shared_secret) || ser<sub>32</sub>(n))''
+*** Compute ''P<sub>n</sub> = B<sub>spend</sub> + t<sub>n</sub>·G''
+*** For each ''output'' in ''outputs_to_check'':
+**** If ''P<sub>n</sub>'' equals ''output'':
+***** Add ''P<sub>n</sub>'' to the wallet
+***** Remove ''output'' from ''outputs_to_check'' and rescan ''outputs_to_check'' with ''n++''
+**** Else, if the wallet has precomputed labels (including the change label, if used)<ref name="precompute_labels">''' Why precompute labels?''' Precomputing the labels is not strictly necessary: a wallet could track the max number of labels it has used (call it ''M'') and scan for labels by adding ''m·G'' to ''P<sub>0</sub>'' for each label ''m'' up to ''M'' and comparing to the transaction outputs. This is more performant than precomputing the labels and checking via subtraction in cases where the number of eligible outputs exceeds the number of labels in use. In practice this will mainly apply to users that choose never to use labels, or users that use a single label for generating silent payment change outputs. If using a large number of labels, the wallet would need to add all possible labels to each output. This ends up being ''n·M'' additions, where ''n'' is the number of outputs in the transaction and ''M'' is the number of labels in the wallet. By precomputing the labels, the wallet only needs to compute ''m·G'' once when creating the labeled address and can determine if a label was used via a lookup, rather than adding each label to each output.</ref>:
+***** Compute ''m·G = output - P<sub>n</sub>''
+***** Check if ''m·G'' exists in the list of labels used by the wallet
+***** If a match is found:
+****** Add the ''P<sub>n</sub> + m·G'' to the wallet
+****** Remove ''output'' from ''outputs_to_check'' and rescan ''outputs_to_check'' with ''n++''
+***** If the label is not found, negate ''output'' and check again
+*** If no matches are found, stop
+
```

Comment:
There is no section for speding outputs, like there is for creating and scanning. While it's not exceedingly difficult to figure out how to calculate the private key of a silent payment output once it's detected in scanning, I think the procedure should be explicitly stated.